Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift

```diff
@@ -472,31 +472,29 @@ open class ChatChannelViewModel: ObservableObject, MessagesDataSource {
     // MARK: - private
     
     private func checkForOlderMessages(index: Int) {
-        if index < channelDataSource.messages.count - 25 {
-            return
-        }
-
+        guard index >= channelDataSource.messages.count - 25 else { return }
+        guard !loadingPreviousMessages else { return }
+        guard !channelController.hasLoadedAllPreviousMessages else { return }
+        
         log.debug("Loading previous messages")
-        if !loadingPreviousMessages {
-            loadingPreviousMessages = true
-            channelDataSource.loadPreviousMessages(
-                before: nil,
-                limit: utils.messageListConfig.pageSize,
-                completion: { [weak self] _ in
-                    guard let self = self else { return }
+        loadingPreviousMessages = true
+        channelDataSource.loadPreviousMessages(
+            before: nil,
+            limit: utils.messageListConfig.pageSize,
+            completion: { [weak self] _ in
+                guard let self = self else { return }
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
We could delete all of the 0.5 second delay logic if we switch to state layer's Chat but that is another topic.